source/math/FloatMatrix.ooc

```diff
@@ -236,6 +236,48 @@ FloatMatrix: cover {
 		this free(Owner Receiver)
 		result
 	}
+	// Thomas algorithm for tridiagonal systems
+	// https://en.wikipedia.org/wiki/Tridiagonal_matrix_algorithm
+	solveTridiagonal: func (y: This) -> This {
+		diagonal := this take() getDiagonal() take()
```

Comment:
I was asked by @simonmika to write a short description of `OwnedBuffer` for people, but I honestly think you're the better person for that. Do you have time? It can (and should be) pretty short, just explaining what goes on in calls like this, what `take()`, `give()` etc. mean and when they're needed.